pkg/workloads/console/runner/runner.go

```diff
@@ -232,11 +228,36 @@ func (c *Runner) Create(ctx context.Context, opts CreateOptions) (*workloadsv1al
 				Hook:       opts.Hook,
 			},
 		)
+		if err != nil {
+			return csl, err
+		}
+
+		return csl, c.exitWithStatus(csl)
 	}
 
 	return csl, nil
 }
 
+func (c *Runner) exitWithStatus(csl *workloadsv1alpha1.Console) error {
+	pod, _, err := c.GetAttachablePod(csl)
+	if err != nil {
+		return err
+	}
+
+	for pod.Status.Phase == "Running" {
```

Comment:
Ah it's not so bad because we'll already have blocked on the `c.Attach` call before this (but not great that this function couldn't be called safely in isolation)

Arguably even then there's an edge case - if the SPDY stream drops for whatever reset (e.g. network blip), and the upstream 'attacher' code doesn't correctly propagate the error, then we could fall into this. Not sure how likely that is though!

Reply:
The reason I did this was more that `pods.get` is not guaranteed to return the most up to date view of the resource, so there may be a delay between Attach returning and the pod appearing to be finished.

I think your edge case is a serious thing to consider though, I have no idea how reliable these SPDY streamers are or if they can 'fail successfully' ... I'll see if I can reimplement this like the other wait func.

Reply:
While adding a watcher into this logic, I discovered that the pod is never (quite) completed when we make this first call to the API to retrieve it so the watcher is very necessary!